### ISSUE TITLE
[BUG] The list height does not update when items size changes with transitions

### ISSUE THREAD

**Issue by BenPrt** [petyosi/react-virtuoso#314]
**Description**
When switching from scrollable list to non-scrollable list because of content change, the list height seems to not be re-calculated. We then have a scrollbar still appearing, even if the list isn't scrollable anymore. When trying to scroll, the list is updated and the scrollbar disappears, ocasionning a weird blinky behaviour.

**Reproduction**
Here is a sandbox illustrating the problem : https://codesandbox.io/s/cool-keldysh-4u7et

**To Reproduce**
Inside the provided sandbox:
1. Expand an accordion from the list : the list now has overflow, so it becomes scrollable.
2. Collapse the same accordion item : the list is back to its initial value, but the scrollbar still is present.
3. Scroll down inside the list : the scrollbar disappears, occasioning a weird re-rendering of the list

**Expected behavior**
When collapsing the accordion item, the list height should be re-calculated and that should make the scrollbar disappear as there is no overflow anymore.

**Additional notes**
Thank you a lot for your work, this package is really great !

**Comment by petyosi** [comment]
Hey @BenPrt ,

Thank you for the repro. I think I have seen this before with the material Accordion. In a nutshell, the accordion animation breaks the item height reporting due to overloading the resize observer. Can you try disabling the animation and see if the problem persists? In general, I would love to be able to support animated expansions, but I would also avoid any complications in that part of the code, since it is fairly easy to break the regular size tracking. 

Cheers,

**Comment by BenPrt** [comment]
Hi @petyosi  !

I just updated the sandbox disabling all MUI transitions, unfortunately it seems to not solve the issue : https://codesandbox.io/s/cool-keldysh-4u7et

Let me know if there is anything I can provide to help you find where the issue can come from !

**Comment by petyosi** [comment]
From what I see in the DOM inspector, some transitions still happen. I tried to find out more about the accordion component, but the MUI docs are quite complicated in that area.

https://user-images.githubusercontent.com/13347/111024924-e060f000-83e9-11eb-94f8-7278e28e0d36.mov

As a reference implementation, check [this simplified sandbox](https://codesandbox.io/s/heuristic-rain-4lc7l?file=/src/App.js). Enabling the transition demonstrates the issue. I guess this ultimately boils down to the ResizeObserver implementation and the way it interprets size changes with animations. 

At this point, I will label this as a limitation of the Virtuoso implementation. If you are willing to explore ways to improve it, the height tracking mechanism happens in [this hook](https://github.com/petyosi/react-virtuoso/blob/master/src/hooks/useSize.ts).

**Comment by kharmony** [comment]
Hi, great work on this tool. I am also having an issue with sizing inside of and MUI accordion. I was able to troubleshoot a bit and identified where the issue is. I added the following to the Virtuoso component and it now shows the data, but it has the position: absolute class.

![image](https://github.com/petyosi/react-virtuoso/assets/85589719/c78df521-723b-4b68-8ce9-da79211a3d38)

![image](https://github.com/petyosi/react-virtuoso/assets/85589719/3f440428-d403-48d5-8334-b378bcd048e8)

However, if I uncheck the position: absolute in the dev tools, it works. Is there anyway we can override that class on the highlighted div?

![image](https://github.com/petyosi/react-virtuoso/assets/85589719/cf991518-e6d6-41b3-9bd9-49cb497a87b9)

Thank you for any assistance you can provide.

**Comment by kharmony** [comment]
Hi @petyosi,

Just to follow-up, I think it is here in the code where this is being assigned.

![image](https://github.com/petyosi/react-virtuoso/assets/85589719/cc0a8435-4af2-483f-943d-53d0d77bbefd)

**Comment by kharmony** [comment]
For anyone that encounters this issue, here is my workaround:

Add this to your makeStyles:
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/480f7dba-944e-466d-89aa-0b9ec29c7cd0)

You can also just hard-code it later in the set state.

Create state:
const [virtuosoClass, setVirtuosoClass] = React.useState();

Create a function:
const handleExpand = (expanded) => {
    if (expanded) setVirtuosoClass(classes.virtuoso);
  };

On the Accordion component add the function for onChange:
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/b3aa4d5e-dc22-4d88-a669-f76e5bff27e1)

Insert a div above the Virtuoso component. Assign the className. Add id. 
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/f9260881-2619-49f0-abec-21b3ac3d3f60)

Basically it doesn't work to just set the class. When the Accordion is opened, the position always gets set to absolute as mentioned above. By doing it on the onChange event, it overrides this. Not sure exactly on the order of operations, but this is the only way I could get it to work.

**Comment by kharmony** [comment]
For anyone that encounters this issue, here is my workaround:

Add this to your makeStyles:
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/480f7dba-944e-466d-89aa-0b9ec29c7cd0)

You can also just hard-code it later in the set state.

Create state:
const [virtuosoClass, setVirtuosoClass] = React.useState();

Create a function:
const handleExpand = (expanded) => {
    if (expanded) setVirtuosoClass(classes.virtuoso);
  };

On the Accordion component add the function for onChange:
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/b3aa4d5e-dc22-4d88-a669-f76e5bff27e1)

Insert a div above the Virtuoso component. Assign the className. Add id. 
![image](https://github.com/petyosi/react-virtuoso/assets/85589719/f9260881-2619-49f0-abec-21b3ac3d3f60)

Basically it doesn't work to just set the class. When the Accordion is opened, the position always gets set to absolute as mentioned above. By doing it on the onChange event, it overrides this. Not sure exactly on the order of operations, but this is the only way I could get it to work.

**Comment by kharmony** [comment]
In the end, none of this works. When you override the position: absolute, it loads the entire data set. I am leaving this here in case it may help anyone else. Think I'll just end up building my own accordion.
